tech-debt: File at 1237 lines exceeds 500-line limit (2.5x); contains 3 separate resolution

## Description

File at 1237 lines exceeds 500-line limit (2.5x); contains 3 separate resolution strategies (module, member, include) plus text-search fallbacks that should be split into focused modules

## Problem

Found at packages/pike-lsp-server/src/features/navigation/definition.ts:1237. Auto-discovered by DGA Architect agent.

## Expected Behavior

See description above.

## Suggested Approach

Follow existing codebase patterns for this type of change.

## Affected Files

packages/pike-lsp-server/src/features/navigation/definition.ts:1237

## Acceptance Criteria

- [ ] Code change implemented
- [ ] Tests added/updated
- [ ] TypeScript compiles: `bun run typecheck`

**Priority:** P0 | **Type:** tech-debt | Auto-discovered

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/core/index.ts(9,50): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(10,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(11,34): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(8,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(166,20): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(206,21): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(212,26): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-actions.ts(25,24): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/inde

--- a/packages/pike-lsp-server/src/features/navigation/definition-utils.ts
+++ b/packages/pike-lsp-server/src/features/navigation/definition-utils.ts
@@ -5,7 +5,7 @@
  * Extracted from definition.ts for testability.
  */
 
-import { TextDocument } from 'vscode-languageserver-textdocument';
+import type { TextDocument } from 'vscode-languageserver-textdocument';
 import type { PikeSymbol } from '@pike-lsp/pike-bridge';
 
 /**

